client/javascript-esm/packages/jolokia/src/jolokia-utils.ts

```diff
@@ -0,0 +1,55 @@
+import {
+  JolokiaErrorResponse,
+  JolokiaSuccessResponse,
+  VersionResponseValue
+} from "./jolokia-types.js"
+
+export type ParseResult<T> = { hasError: false; parsed: T } | { hasError: true; error: string }
+
+function isObject(value: unknown): value is object {
+  const type = typeof value
+  return value != null && (type === 'object' || type === 'function')
+}
+
+export function isResponseSuccessType(o: unknown): o is JolokiaSuccessResponse {
+  return isObject(o) && 'status' in o && 'timestamp' in o && 'value' in o
+}
+
+export function isResponseErrorType(o: unknown): o is JolokiaErrorResponse {
+  return isObject(o) && 'error_type' in o && 'error' in o
+}
+
+export function isVersionResponseType(o: unknown): o is VersionResponseValue {
```

Comment:
I think we can extend the main Jolokia object from `jolokia-simple.ts` for the function. And again the name seems a bit redundant, maybe we can name it `isVersionResponse(o: unknown)`.